base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,37 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+    WCHAR szExplorer[] = L"explorer";
+	DWORD dwType;
+	DWORD dwBufferSize = sizeof(szShell);
+
+	if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+		0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        return FALSE;
+    }
```

Comment:
Should the default be TRUE here? If you can't read this key the registry is in a broken state?

Reply:
I was debating on whether to have that TRUE or FALSE for the failsafe mode. If it's preferred to be TRUE then I will change it.

Reply:
It probably makes more sense to assume it's the shell rather than not, so I'll do this.